### PR TITLE
Add GitHub Actions building

### PR DESCRIPTION
**Pull request by gauntl3t12**
Removes the requirement on you to build and release the containers

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      uses: actions/checkout@v3
+    - name: Build the Docker image
+      run: docker build . --build-arg POSTGRES_VERSION=${{ matrix.version }} -t brconn98/pg_dump:${{ matrix.version }} --file Dockerfile
+    - name: Log into docker.io
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
+      with:
+        registry: docker.io
+        username: martlark
+        password: ${{ secrets.DOCKER_PAT }}
```
Comment (gauntl3t12) [inline .github/workflows/docker-image.yml:29]: The [DOCKER_PAT needs to be obtained](https://www.docker.com/blog/docker-hub-new-personal-access-tokens/) by @Martlark and [set in the settings of the project](https://docs.github.com/en/actions/security-guides/encrypted-secrets).

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+
+  bld-and-push:
+    strategy:
+      matrix:
+        version: [9.6, 10.17, 11.12, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 12.10, 13.0, 13.1, 13.6, 14.2, latest]
```
Comment (gauntl3t12) [inline .github/workflows/docker-image.yml:16]: New versions can be added here at any time

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (Martlark) [reply .github/workflows/docker-image.yml:29]: nice work!

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Build the Docker image
+      run: docker build . --build-arg POSTGRES_VERSION=${{ matrix.version }} -t brconn98/pg_dump:${{ matrix.version }} --file Dockerfile
```
Comment (Martlark) [inline .github/workflows/docker-image.yml]: tag should be martlark/pg_dump

**Review by Martlark — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-image.yml (hunk 3 above)
Comment (gauntl3t12) [reply .github/workflows/docker-image.yml]: Realized the issue spot. Fixed.

**Review by Martlark — APPROVED** [review]

**Comment by gauntl3t12** [comment]
Should be good to merge whenever @Martlark is all set

**Comment by gauntl3t12** [comment]
Any movement on merging?

**Pull request merged**